geomstats/geometry/positive_lower_triangular_matrices.py

```diff
@@ -0,0 +1,359 @@
+        aux = Matrices.to_lower_triangular_diagonal_scaled(
+            Matrices.mul(inv_base_point, tanget_vec, inv_transpose_base_point)
+        )
+        inverse_differential_gram = Matrices.mul(base_point, aux)
+        return inverse_differential_gram
+
+
+class CholeskyMetric(RiemannianMetric):
+    """Class for the cholesky metric on the cholesky space."""
+
+    def __init__(self, n):
+        """Build the CholeskyMetric
+
+        Parameters
+        ----------
+        n : int
+            Integer representing the shape of the matrices: n x n.
+
+
+        References
+        ----------
+        .. [TP2019] . "Riemannian Geometry of Symmetric
+        Positive Definite Matrices Via Cholesky Decomposition"
+        SIAM journal on Matrix Analysis and Applications , 2019.
+         https://arxiv.org/abs/1908.09326
+        """
+        dim = int(n * (n + 1) / 2)
+        super(CholeskyMetric, self).__init__(
+            dim=dim, signature=(dim, 0), default_point_type="matrix"
+        )
+        self.n = n
+
+    @staticmethod
+    def diag_inner_product(tangent_vec_a, tangent_vec_b, base_point):
+        """Compute the inner product using only diagonal elements.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        tangent_vec_b : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        ip_diagonal : array-like, shape=[...]
+            Inner-product.
+        """
+        inv_sqrt_diagonal = gs.power(Matrices.diagonal(base_point), -2)
+        tangent_vec_a_diagonal = Matrices.diagonal(tangent_vec_a)
+        tangent_vec_b_diagonal = Matrices.diagonal(tangent_vec_b)
+        prod = tangent_vec_a_diagonal * tangent_vec_b_diagonal * inv_sqrt_diagonal
+        ip_diagonal = gs.sum(prod, axis=-1)
+        return ip_diagonal
+
+    @staticmethod
+    def strictly_lower_inner_product(tangent_vec_a, tangent_vec_b):
+        """Compute the inner product using only strictly lower triangular elements.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        tangent_vec_b : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+
+        Returns
+        -------
+        ip_sl : array-like, shape=[...]
+            Inner-product.
+        """
+        sl_tagnet_vec_a = gs.tril_to_vec(tangent_vec_a, k=-1)
+        sl_tagnet_vec_b = gs.tril_to_vec(tangent_vec_b, k=-1)
+        ip_sl = gs.einsum("...i,...i->...", sl_tagnet_vec_a, sl_tagnet_vec_b)
+        return ip_sl
+
+    @classmethod
+    def inner_product(cls, tangent_vec_a, tangent_vec_b, base_point):
+        """Compute the inner product using only strictly lower triangular elements.
```

Comment:
Ditto

Reply:
nope. It is inner product!. (it satisfies symmetric, non-degenerate, positive bi-linear conditions )

Reply:
Adapt the docstring

Reply:
(the copy-pasting has had a problem)